src/app/services/article.service.ts

```diff
@@ -28,106 +53,21 @@ export class ArticleService {
-  // Call the function to load articles  
-  stripHtml(html: string): string {
-    const div = document.createElement('div');
-    div.innerHTML = html;
-    return div.textContent || div.innerText || '';
-  }
-  
-  getArticleList(): Observable<Article[]> {
-    // Use 'from' to convert the Promise returned by loadArticles to an Observable
-    return from(this.loadArticles()).pipe(
-      // Switch to returning this.articles if it's available after loading
-      switchMap(() => {
-        if (this.articles?.length) {
-          return of(this.articles); // Return the articles if already loaded
-        }
-  
-        // Fallback to hardcoded articles after delay if this.articles is empty
-        return of(getHardcodedArticles()).pipe(
-          delay(1500), // Simulate delay (e.g., loading time)
-          tap((articles: Article[]) => this.articles = articles) // Update this.articles
-        );
-      })
-    );
-  }
-  readOne(articleId: string): Observable<Article> {
-    return of(getHardcodedArticles().find((article: Article) => article.articleId === articleId) || {
-      header: 'invalid-article',
-      body: 'invalid-article',
-      imageURI: 'none',
-      subCategory: 'Article',
-      articleId: 'invalid-article'
-    });
-  }
-
-  saveArticleBody = async (article:Article) => {
-    console.log(`GET articleId: ${article.articleId}`);
-    try {
-      const documentId = article.meta?.documentId;
-      const { body, header, imageURI, meta } = article;
-      const documentName = article.meta?.name;
-      const category = meta?.category;
-      const subCategory = meta?.subCategory;
-
-      const response = await fetch(`https://firestore.googleapis.com/v1/${documentName}?updateMask.fieldPaths=body&updateMask.fieldPaths=title`, {
-        method: 'PATCH',
-        headers: {
-          'Content-Type': 'application/json'
-        },
-        body: JSON.stringify({
-          "fields": {
-            "body": {
-              "stringValue": body
-            },
-            "header": {
-              "stringValue": header
-            },
-            "imageURI": {
-              "stringValue": imageURI
-            },
-            "meta": {
-              "mapValue": {
-                "fields": {
-                  "category": {
-                    "stringValue": category
-                  },
-                  "subCategory": {
-                    "stringValue": subCategory
-                  }
-                }
-              }
-            }
-          }
-        }
-      )
-      });
-
-      if (!response.ok) {
-        throw new Error(`HTTP error! Status: ${response.status}`);
-      }
-      await response.json();
 
     } catch (error) {
-      console.error('Error fetching articles:', error);
+      console.error("Error loading articles", error);
     }
-  }  
+    return this.articles;
+  };
 
-  getArticle = async (articleId: string) => {
+  private async getArticle(articleId: string): Promise<Article | null> {
```

Comment:
we can call this `fetchArticle` if we want a separate `getArticle`.

Reply:
Naming it: fetchArticleFromFirestore() for now just to get the PR in, can change later.